manager/integration/tests/test_basic.py

```diff
@@ -4438,3 +4439,128 @@ def test_space_usage_for_rebuilding_only_volume():  # NOQA
+    4.   Wait and check the backup failed
+    5.   Wait and check if the backup was not deleted.
+    6.   Cleanup
+    """
+    bt_poll_interval = "60"
+    failed_backup_ttl = "3"
+    volume_size = 256 * Mi
+
+    # set backupstore poll interval to 60 sec
+    set_backupstore_poll_interval(client, bt_poll_interval)
+
+    # set failed backup time to live to 3 min
+    failed_backup_ttl_setting = client.by_id_setting(SETTING_FAILED_BACKUP_TTL)
+    new_failed_backup_ttl_setting = client.update(
+        failed_backup_ttl_setting, value=failed_backup_ttl)
+    assert new_failed_backup_ttl_setting.value == failed_backup_ttl
+
+    backupstore_cleanup(client)
+
+    # create a volume
+    vol = create_and_check_volume(client, volume_name,
+                                  num_of_replicas=2, size=str(volume_size))
+
+    # attach the volume
+    vol.attach(hostId=get_self_host_id())
+    vol = wait_for_volume_healthy(client, volume_name)
+
+    # create a snapshot and a successful backup
+    snap = create_snapshot(client, volume_name)
+    vol.snapshotBackup(name=snap.name)
+
+    # write some data to the volume
+    data = {
+        'pos': 0,
+        'content': common.generate_random_data(volume_size),
+    }
+    write_volume_data(vol, data)
+
+    # create a snapshot and a backup
+    snap = create_snapshot(client, volume_name)
+    vol.snapshotBackup(name=snap.name)
+
+    # check backup status is in an InProgress state
+    _, backup = find_backup(client, volume_name, snap.name)
+    backup_id = backup.id
+
+    def backup_inprogress_predicate(b):
+        return b.id == backup_id and "InProgress" in b.state
+    common.wait_for_backup_state(client, volume_name,
+                                 backup_inprogress_predicate)
+
+    # crash all replicas of the volume
+    crash_replica_processes(client, core_api, volume_name)
+
+    # backup status should be in an Error state and with an error message
+    def backup_failure_predicate(b):
+        return b.id == backup_id and "Error" in b.state and b.error != ""
+    common.wait_for_backup_state(client, volume_name,
+                                 backup_failure_predicate)
+
+    # wait in 5 minutes for automatic failed backup cleanup
+    wait_for_backup_delete(client, volume_name, backup.name)
+
+    # disable failed backup auto cleanup
+    failed_backup_ttl = "0"
+    failed_backup_ttl_setting = client.by_id_setting(SETTING_FAILED_BACKUP_TTL)
+    new_failed_backup_ttl_setting = client.update(
+        failed_backup_ttl_setting, value=failed_backup_ttl)
+    assert new_failed_backup_ttl_setting.value == failed_backup_ttl
+
+    # create a snapshot and a backup
+    snap = create_snapshot(client, volume_name)
+    vol = client.by_id_volume(volume_name)
+    vol.snapshotBackup(name=snap.name)
+
+    # check backup status is in an InProgress state
+    _, backup = find_backup(client, volume_name, snap.name)
+    backup_id = backup.id
+
+    def backup_inprogress_predicate(b):
```

Comment:
Can we separate this to two test cases/methods instead? you can use one as a basic test method, but the other just base on it.

Reply:
Do you mean the enable and disable cases?

Corrected the sign-off email, thanks.

Reply:
The code look duplicated. You can have a function(for example, backup_disable_auto_cleanup_test) to cover the duplicated code and use it in both cases.

Reply:
Yeah, modified, moved the duplicated code into a method `backup_failed_cleanup` and both enabling and disabling test case use the method.
Thanks.